### ISSUE TITLE
Build error

### ISSUE THREAD

**Issue by marcomasa** [robocup-at-work/atwork-commander#23]
Ubuntu 16.04 & Kinetic

```
Errors     << atwork_commander:make /home/marco/catkin_ws/logs/atwork_commander/build.make.002.log                                                                                   
CMakeFiles/control.dir/src/control_node.cpp.o: In Funktion `parseArgs(int, char**)':
/home/marco/catkin_ws/src/ohmnibot/atwork/atwork_league/atwork-commander/atwork_commander/src/control_node.cpp:274: Nicht definierter Verweis auf `boost::program_options::options_description::m_default_line_length'
/home/marco/catkin_ws/src/ohmnibot/atwork/atwork_league/atwork-commander/atwork_commander/src/control_node.cpp:274: Nicht definierter Verweis auf `boost::program_options::options_description::m_default_line_length'
/home/marco/catkin_ws/src/ohmnibot/atwork/atwork_league/atwork-commander/atwork_commander/src/control_node.cpp:274: Nicht definierter Verweis auf `boost::program_options::options_description::options_description(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int, unsigned int)'
/home/marco/catkin_ws/src/ohmnibot/atwork/atwork_league/atwork-commander/atwork_commander/src/control_node.cpp:276: Nicht definierter Verweis auf `boost::program_options::options_description::add_options()'
/home/marco/catkin_ws/src/ohmnibot/atwork/atwork_league/atwork-commander/atwork_commander/src/control_node.cpp:277: Nicht definierter Verweis auf `boost::program_options::options_description_easy_init::operator()(char const*, boost::program_options::value_semantic const*, char const*)'
/home/marco/catkin_ws/src/ohmnibot/atwork/atwork_league/atwork-commander/atwork_commander/src/control_node.cpp:278: Nicht definierter Verweis auf `boost::program_options::options_description_easy_init::operator()(char const*, boost::program_options::value_semantic const*, char const*)'
/home/marco/catkin_ws/src/ohmnibot/atwork/atwork_league/atwork-commander/atwork_commander/src/control_node.cpp:279: Nicht definierter Verweis auf `boost::program_options::options_description_easy_init::operator()(char const*, boost::program_options::value_semantic const*, char const*)'
/home/marco/catkin_ws/src/ohmnibot/atwork/atwork_league/atwork-commander/atwork_commander/src/control_node.cpp:281: Nicht definierter Verweis auf `boost::program_options::options_description_easy_init::operator()(char const*, char const*)'
/home/marco/catkin_ws/src/ohmnibot/atwork/atwork_league/atwork-commander/atwork_commander/src/control_node.cpp:282: Nicht definierter Verweis auf `boost::program_options::positional_options_description::positional_options_description()'
/home/marco/catkin_ws/src/ohmnibot/atwork/atwork_league/atwork-commander/atwork_commander/src/control_node.cpp:283: Nicht definierter Verweis auf `boost::program_options::positional_options_description::add(char const*, int)'
/home/marco/catkin_ws/src/ohmnibot/atwork/atwork_league/atwork-commander/atwork_commander/src/control_node.cpp:283: Nicht definierter Verweis auf `boost::program_options::positional_options_description::add(char const*, int)'
/home/marco/catkin_ws/src/ohmnibot/atwork/atwork_league/atwork-commander/atwork_commander/src/control_node.cpp:285: Nicht definierter Verweis auf `boost::program_options::variables_map::variables_map()'
/home/marco/catkin_ws/src/ohmnibot/atwork/atwork_league/atwork-commander/atwork_commander/src/control_node.cpp:286: Nicht definierter Verweis auf `boost::program_options::store(boost::program_options::basic_parsed_options<char> const&, boost::program_options::variables_map&, bool)'
/home/marco/catkin_ws/src/ohmnibot/atwork/atwork_league/atwork-commander/atwork_commander/src/control_node.cpp:287: Nicht definierter Verweis auf `boost::program_options::notify(boost::program_options::variables_map&)'
/home/marco/catkin_ws/src/ohmnibot/atwork/atwork_league/atwork-commander/atwork_commander/src/control_node.cpp:298: Nicht definierter Verweis auf `boost::program_options::operator<<(std::ostream&, boost::program_options::options_description const&)'
CMakeFiles/control.dir/src/control_node.cpp.o: In Funktion `boost::program_options::error_with_option_name::~error_with_option_name()':
/usr/include/boost/program_options/errors.hpp:119: Nicht definierter Verweis auf `vtable for boost::program_options::error_with_option_name'
CMakeFiles/control.dir/src/control_node.cpp.o: In Funktion `boost::program_options::value_semantic_codecvt_helper<char>::value_semantic_codecvt_helper()':
/usr/include/boost/program_options/value_semantic.hpp:91: Nicht definierter Verweis auf `vtable for boost::program_options::value_semantic_codecvt_helper<char>'
CMakeFiles/control.dir/src/control_node.cpp.o: In Funktion `boost::program_options::value_semantic_codecvt_helper<char>::~value_semantic_codecvt_helper()':
/usr/include/boost/program_options/value_semantic.hpp:91: Nicht definierter Verweis auf `vtable for boost::program_options::value_semantic_codecvt_helper<char>'
CMakeFiles/control.dir/src/control_node.cpp.o: In Funktion `boost::program_options::basic_command_line_parser<char>::basic_command_line_parser(int, char const* const*)':
/usr/include/boost/program_options/detail/parsers.hpp:43: Nicht definierter Verweis auf `boost::program_options::detail::cmdline::cmdline(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)'
CMakeFiles/control.dir/src/control_node.cpp.o: In Funktion `boost::program_options::basic_command_line_parser<char>::options(boost::program_options::options_description const&)':
/usr/include/boost/program_options/detail/parsers.hpp:51: Nicht definierter Verweis auf `boost::program_options::detail::cmdline::set_options_description(boost::program_options::options_description const&)'
CMakeFiles/control.dir/src/control_node.cpp.o: In Funktion `boost::program_options::basic_command_line_parser<char>::positional(boost::program_options::positional_options_description const&)':
/usr/include/boost/program_options/detail/parsers.hpp:61: Nicht definierter Verweis auf `boost::program_options::detail::cmdline::set_positional_options(boost::program_options::positional_options_description const&)'
CMakeFiles/control.dir/src/control_node.cpp.o: In Funktion `boost::program_options::basic_command_line_parser<char>::run()':
/usr/include/boost/program_options/detail/parsers.hpp:107: Nicht definierter Verweis auf `boost::program_options::detail::cmdline::get_canonical_option_prefix()'
/usr/include/boost/program_options/detail/parsers.hpp:108: Nicht definierter Verweis auf `boost::program_options::detail::cmdline::run()'
CMakeFiles/control.dir/src/control_node.cpp.o: In Funktion `std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > boost::program_options::to_internal<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)':
/usr/include/boost/program_options/detail/convert.hpp:79: Nicht definierter Verweis auf `boost::program_options::to_internal(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
CMakeFiles/control.dir/src/control_node.cpp.o:(.rodata._ZTVN5boost15program_options11typed_valueIbcEE[_ZTVN5boost15program_options11typed_valueIbcEE]+0x38): Nicht definierter Verweis auf `boost::program_options::value_semantic_codecvt_helper<char>::parse(boost::any&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, bool) const'
CMakeFiles/control.dir/src/control_node.cpp.o:(.rodata._ZTVN5boost15program_options11typed_valueISt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS8_EEcEE[_ZTVN5boost15program_options11typed_valueISt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS8_EEcEE]+0x38): Nicht definierter Verweis auf `boost::program_options::value_semantic_codecvt_helper<char>::parse(boost::any&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, bool) const'
CMakeFiles/control.dir/src/control_node.cpp.o:(.rodata._ZTVN5boost15program_options11typed_valueINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEcEE[_ZTVN5boost15program_options11typed_valueINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEcEE]+0x38): Nicht definierter Verweis auf `boost::program_options::value_semantic_codecvt_helper<char>::parse(boost::any&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, bool) const'
CMakeFiles/control.dir/src/control_node.cpp.o:(.rodata._ZTVN5boost15program_options20invalid_option_valueE[_ZTVN5boost15program_options20invalid_option_valueE]+0x20): Nicht definierter Verweis auf `boost::program_options::error_with_option_name::what() const'
CMakeFiles/control.dir/src/control_node.cpp.o:(.rodata._ZTVN5boost15program_options20invalid_option_valueE[_ZTVN5boost15program_options20invalid_option_valueE]+0x30): Nicht definierter Verweis auf `boost::program_options::error_with_option_name::substitute_placeholders(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
CMakeFiles/control.dir/src/control_node.cpp.o:(.rodata._ZTVN5boost15program_options16validation_errorE[_ZTVN5boost15program_options16validation_errorE]+0x20): Nicht definierter Verweis auf `boost::program_options::error_with_option_name::what() const'
CMakeFiles/control.dir/src/control_node.cpp.o:(.rodata._ZTVN5boost15program_options16validation_errorE[_ZTVN5boost15program_options16validation_errorE]+0x30): Nicht definierter Verweis auf `boost::program_options::error_with_option_name::substitute_placeholders(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
CMakeFiles/control.dir/src/control_node.cpp.o:(.rodata._ZTIN5boost15program_options11typed_valueIbcEE[_ZTIN5boost15program_options11typed_valueIbcEE]+0x18): Nicht definierter Verweis auf `typeinfo for boost::program_options::value_semantic_codecvt_helper<char>'
CMakeFiles/control.dir/src/control_node.cpp.o:(.rodata._ZTIN5boost15program_options11typed_valueISt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS8_EEcEE[_ZTIN5boost15program_options11typed_valueISt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS8_EEcEE]+0x18): Nicht definierter Verweis auf `typeinfo for boost::program_options::value_semantic_codecvt_helper<char>'
CMakeFiles/control.dir/src/control_node.cpp.o:(.rodata._ZTIN5boost15program_options11typed_valueINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEcEE[_ZTIN5boost15program_options11typed_valueINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEcEE]+0x18): Nicht definierter Verweis auf `typeinfo for boost::program_options::value_semantic_codecvt_helper<char>'
CMakeFiles/control.dir/src/control_node.cpp.o:(.rodata._ZTIN5boost15program_options16validation_errorE[_ZTIN5boost15program_options16validation_errorE]+0x10): Nicht definierter Verweis auf `typeinfo for boost::program_options::error_with_option_name'
CMakeFiles/control.dir/src/control_node.cpp.o: In Funktion `boost::program_options::variables_map::~variables_map()':
/usr/include/boost/program_options/variables_map.hpp:146: Nicht definierter Verweis auf `vtable for boost::program_options::variables_map'
CMakeFiles/control.dir/src/control_node.cpp.o: In Funktion `boost::program_options::typed_value<bool, char>::name() const':
/usr/include/boost/program_options/detail/value_semantic.hpp:19: Nicht definierter Verweis auf `boost::program_options::arg[abi:cxx11]'
CMakeFiles/control.dir/src/control_node.cpp.o: In Funktion `boost::program_options::typed_value<bool, char>::xparse(boost::any&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&) const':
/usr/include/boost/program_options/detail/value_semantic.hpp:167: Nicht definierter Verweis auf `boost::program_options::validate(boost::any&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, bool*, int)'
CMakeFiles/control.dir/src/control_node.cpp.o: In Funktion `boost::program_options::typed_value<std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, char>::name() const':
/usr/include/boost/program_options/detail/value_semantic.hpp:19: Nicht definierter Verweis auf `boost::program_options::arg[abi:cxx11]'
CMakeFiles/control.dir/src/control_node.cpp.o: In Funktion `boost::program_options::typed_value<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, char>::name() const':
/usr/include/boost/program_options/detail/value_semantic.hpp:19: Nicht definierter Verweis auf `boost::program_options::arg[abi:cxx11]'
CMakeFiles/control.dir/src/control_node.cpp.o: In Funktion `boost::program_options::typed_value<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, char>::xparse(boost::any&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&) const':
/usr/include/boost/program_options/detail/value_semantic.hpp:167: Nicht definierter Verweis auf `boost::program_options::validate(boost::any&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, int)'
CMakeFiles/control.dir/src/control_node.cpp.o: In Funktion `void boost::program_options::validate<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, char>(boost::any&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >*, int)':
/usr/include/boost/program_options/detail/value_semantic.hpp:146: Nicht definierter Verweis auf `boost::program_options::validate(boost::any&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, int)'
/usr/include/boost/program_options/detail/value_semantic.hpp:150: Nicht definierter Verweis auf `boost::program_options::invalid_option_value::invalid_option_value(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
CMakeFiles/control.dir/src/control_node.cpp.o: In Funktion `boost::program_options::error_with_option_name::error_with_option_name(boost::program_options::error_with_option_name const&)':
/usr/include/boost/program_options/errors.hpp:89: Nicht definierter Verweis auf `vtable for boost::program_options::error_with_option_name'
CMakeFiles/control.dir/src/control_node.cpp.o:(.rodata._ZTVN5boost16exception_detail10clone_implINS0_19error_info_injectorINS_15program_options20invalid_option_valueEEEEE[_ZTVN5boost16exception_detail10clone_implINS0_19error_info_injectorINS_15program_options20invalid_option_valueEEEEE]+0x28): Nicht definierter Verweis auf `boost::program_options::error_with_option_name::what() const'
CMakeFiles/control.dir/src/control_node.cpp.o:(.rodata._ZTVN5boost16exception_detail10clone_implINS0_19error_info_injectorINS_15program_options20invalid_option_valueEEEEE[_ZTVN5boost16exception_detail10clone_implINS0_19error_info_injectorINS_15program_options20invalid_option_valueEEEEE]+0x38): Nicht definierter Verweis auf `boost::program_options::error_with_option_name::substitute_placeholders(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
CMakeFiles/control.dir/src/control_node.cpp.o:(.rodata._ZTVN5boost16exception_detail19error_info_injectorINS_15program_options20invalid_option_valueEEE[_ZTVN5boost16exception_detail19error_info_injectorINS_15program_options20invalid_option_valueEEE]+0x20): Nicht definierter Verweis auf `boost::program_options::error_with_option_name::what() const'
CMakeFiles/control.dir/src/control_node.cpp.o:(.rodata._ZTVN5boost16exception_detail19error_info_injectorINS_15program_options20invalid_option_valueEEE[_ZTVN5boost16exception_detail19error_info_injectorINS_15program_options20invalid_option_valueEEE]+0x30): Nicht definierter Verweis auf `boost::program_options::error_with_option_name::substitute_placeholders(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
collect2: error: ld returned 1 exit status
make[2]: *** [/home/marco/catkin_ws/devel/.private/atwork_commander/lib/atwork_commander/control] Fehler 1
make[1]: *** [CMakeFiles/control.dir/all] Fehler 2
make: *** [all] Fehler 2
```

**Comment by marcomasa** [comment]
Is this the same issue as #22 ?

**Comment by steup** [comment]
Yes it is!

**Comment by steup** [comment]
Probably an LTO issue. WIll investigate.

**Comment by steup** [comment]
Should be fixed through the fixing of #22. Please check out d1d334847ce0aeb521952d54d9a5ba54fa2f6d76 or later and test.

Closing for now.
